### ISSUE TITLE
Unable to build the library

### ISSUE THREAD

**Issue by Abhyant** [microsoft/msquic#1064]
### Describe the bug
When trying to build the library, I'm getting the following error

`Start-Process: C:\msquic-main\scripts\build.ps1:238                                                                     Line |                                                                                                                   238 |  …  $process = Start-Process cmake $Arguments -PassThru -NoNewWindow -Wo …                                            |                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                                              | This command cannot be run due to the error: The system cannot find the file specified.`

I am following the instructions provided in the build document. I have already tried changing the execution policy which helped resolve other errors but this one still persists.

### Steps to reproduce the behavior

1. Run `./scripts/build.ps1` from the parent directory
2. View output
3. See error

System Configuration-
Windows Server 2019 Build 20262.1000
Visual Studio 2019 Community ver-16.8.2
Powershell 7.1.0
.NET SDK 5.0.1
.NET Core 3.1
Cmake 3.19.0
Installed dependencies from the `prepare-machine.ps1 -Configuration Dev` successfully.

**Comment by ThadHouse** [comment]
This looks like cmake did not get added to PATH when you installed it. From a terminal, are you able to just run `cmake` ?

**Comment by Abhyant** [comment]
@ThadHouse Thanks as you stated, cmake was not added to the PATH and adding it did resolve the issue, but it ran into another error 
`Write-Error: C:\msquic-main\scripts\build.ps1:312                                                                       Line |                                                                                                                   312 |      CMake-Execute $Arguments                                                                                         |      ~~~~~~~~~~~~~~~~~~~~~~~~                                                                                         | [11/26/2020 01:03:35] CMake exited with status code 1 `

**Comment by nibanks** [comment]
@abhyant is that the only output? It still seems to indicate a problem with cmake.

**Comment by Abhyant** [comment]
@nibanks Here's the complete output of the build command

`[11/26/2020 02:26:43] Generating files...  ` 
 `[11/26/2020 02:26:43] cmake -g 'Visual Studio 16 2019' -A x64 -DQUIC_TLS=schannel -DQUIC_OUTPUT_DIR=C:\msquic-main\artifacts\bin\windows\x64_Release_schannel -DQUIC_ENABLE_LOGGING=on ../../..                                                 -- CMAKE Version: 3.19.1                                                                                                -- Source Dir: C:/msquic-main                                                                                           -- Host System name: Windows                                                                                            -- Setting policy 0091                                                                                                  -- System name: Windows                                                                                                 -- System version: 10.0.18362.0                                                                                         -- Platform version: 10.0.18362.0                                                                                       -- Build type:                                                                                                          -- QUIC Platform: windows                                                                                               CMake Error at cmake/GitCommands.cmake:62 (message):                                                                      message called with incorrect number of arguments                                                                     Call Stack (most recent call first):                                                                                      cmake/SourceLink.cmake:41 (get_git_remote_url)                                                                          CMakeLists.txt:149 (source_link)`

`CMake Error at cmake/GitCommands.cmake:63 (message):                                                                      Failed to get C:/msquic-main git remote                                                                               Call Stack (most recent call first):                                                                                      cmake/SourceLink.cmake:41 (get_git_remote_url)                                                                          CMakeLists.txt:149 (source_link)                                                                                                                                                                                                                                                                                                                                      -- Configuring incomplete, errors occurred!                                                                             See also "C:/msquic-main/build/windows/x64_schannel/CMakeFiles/CMakeOutput.log".                                        Write-Error: C:\msquic-main\scripts\build.ps1:312                                                                       Line |                                                                                                                   312 |      CMake-Execute $Arguments                                                                                         |      ~~~~~~~~~~~~~~~~~~~~~~~~                                                                                         | [11/26/2020 02:26:43] CMake exited with status code 1`

Apologies if this is a stupid error, haven't used cmake a lot in the past.

**Comment by ThadHouse** [comment]
Are you trying to build from just downloading the zip from GitHub, rather then cloning as a Git repo? The setup does require the repo to be cloned, and git installed in path, rather then just downloading the zip.

This is something I can look at fixing next week so that's not a hard requirement, but that's how you can solve it now.

**Comment by Abhyant** [comment]
@ThadHouse So I cloned the repo and after running the build command I got the following 

`CMake Error at CMakeLists.txt:598 (add_subdirectory):
  The source directory C:/msquic/submodules/googletest
  does not contain a CMakeLists.txt file.`
`CMake Error at CMakeLists.txt:600 (set_property):
  set_property could not find TARGET gtest.  Perhaps it has not yet been
  created.`
`CMake Error at CMakeLists.txt:601 (set_property):                                                                         set_property could not find TARGET gtest_main.  Perhaps it has not yet been                                             created.`
`-- Configuring incomplete, errors occurred!                                                                             See also "C:/msquic/build/windows/x64_schannel/CMakeFiles/CMakeOutput.log".                                             See also "C:/msquic/build/windows/x64_schannel/CMakeFiles/CMakeError.log". `       
                                      `←[91mWrite-Error: ←[0mC:\msquic\scripts\build.ps1:310         `    
                                                    `←[96mLine |`
`←[96m 310 | ←[0m     ←[96mCMake-Execute $Arguments←[0m`    
                                                              `←[96m     | ←[91m     ~~~~~~~~~~~~~~~~~~~~~~~~  `  
                                                                      `←[91m←[96m     | ←[91m[11/26/2020 22:11:47] CMake exited with status code 1`  
                                           `←[0m`
